test/fiskaltrust.Launcher.IntegrationTest/Plebian/PlebianTests.cs

```diff
@@ -0,0 +1,145 @@
+                Package = "test",
+                Url = new[] { "grpc://localhost:1502", "rest://localhost:1503" },
+                Version = "1.0.0"
+            };
+
+            var pos = Mock.Of<IPOS>();
+
+            Mock.Get(pos).Setup(x => x.EchoAsync(It.IsAny<EchoRequest>())).ReturnsAsync((EchoRequest r) =>
+            {
+                return new EchoResponse { Message = r.Message };
+            });
+
+            await RunTest(pos, PackageType.Queue, packageConfiguration, async () =>
+            {
+                var grpcClient = new POSClientFactory().CreateClient(new ClientConfiguration
+                {
+                    Url = packageConfiguration.Url[0],
+                    UrlType = "grpc",
+                    RetryCount = null
+                });
+
+                (await grpcClient.EchoAsync(new EchoRequest { Message = "test" })).Should().Match<EchoResponse>(r => r.Message == "test");
+
+                var restClient = new POSClientFactory().CreateClient(new ClientConfiguration
+                {
+                    Url = packageConfiguration.Url[1],
+                    UrlType = "rest",
+                    RetryCount = null
+                });
+
+                (await restClient.EchoAsync(new EchoRequest { Message = "test" })).Should().Match<EchoResponse>(r => r.Message == "test");
+
+            });
+        }
+
+        private static async Task RunTest<T>(T instance, PackageType packageType, PackageConfiguration packageConfiguration, Func<Task> checks) where T : class
+        {
+            var launcherConfiguration = new LauncherConfiguration
+            {
+                CashboxId = Guid.NewGuid(),
+                ServiceFolder = "TestService",
+                LogLevel = LogLevel.Debug
+            };
+
+            launcherConfiguration.EnableDefaults();
+
+            var plebianConfiguration = new PlebianConfiguration
+            {
+                PackageId = packageConfiguration.Id,
+                PackageType = packageType
+            };
+
+            var started = new TaskCompletionSource();
+            var monarch = Mock.Of<IProcessHostMonarch>();
+            Mock.Get(monarch).Setup(x => x.Started()).Callback(() => started.SetResult());
+            var processHostService = new ProcessHostService(new Dictionary<Guid, IProcessHostMonarch> { { packageConfiguration.Id, monarch } }, Mock.Of<Serilog.ILogger>());
+
+            var services = new ServiceCollection();
+
+            services.AddSingleton(_ => instance);
+
+            var lifetime = new HostApplicationLifetime();
+
+            var hostingService = new HostingService(Mock.Of<ILogger<HostingService>>(), packageConfiguration, launcherConfiguration, processHostService);
+            var plebian = new ProcessHostPlebian(Mock.Of<ILogger<ProcessHostPlebian>>(), hostingService, launcherConfiguration, packageConfiguration, plebianConfiguration, services.BuildServiceProvider(), lifetime, processHostService);
```

Comment:
I think we need to wrap this in a `using`, otherwise it may keep running in the background 🤔

Reply:
I'm not sure what you mean 😅 none of those are disposable

Reply:
`ProcessHostPlebian` should be `IDisposable`, right?`
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.backgroundservice?view=dotnet-plat-ext-7.0